### PR TITLE
tools: db migrate down functionality

### PR DESCRIPTION
**Pull request by mcutalo88**
### Description
Here we add the ability to migrate down, a single invocation of the down migration will only ever do one version back. Given how dangerous this feature is as it can lead to irrevocable data loss I felt that this was good enough.

### Testing Performed

Starting on the latest migration version which is currently `6`
```
clutch=# \dt
              List of relations
 Schema |       Name        | Type  | Owner
--------+-------------------+-------+--------
 public | audit_events      | table | clutch
 public | experiment_config | table | clutch
 public | experiment_run    | table | clutch
 public | schema_migrations | table | clutch
 public | topology_cache    | table | clutch
(5 rows)

clutch=# select * from schema_migrations;
 version | dirty
---------+-------
       6 | f
(1 row)
````

Running migrate with the down flag, should only bring us back one version to `5`.
```
➜  go run migrate.go -d -c ./testdata/clutch-config-test.yaml
2020-12-02T10:58:28.452-0800	INFO	migrate/migrate.go:129	Utilizing migration directory:	{"migrationDir": "/Users/mcutalo/gocode/src/github.com/lyft/clutch/backend/cmd/migrate/migrations"}
2020-12-02T10:58:28.455-0800	INFO	migrate/migrate.go:93	using database	{"hostInfo": "clutch@0.0.0.0:5432"}
2020-12-02T10:58:28.455-0800	WARN	migrate/migrate.go:95	Migrating DOWN by ONE version from (6 -> 5) this migration has the potential to cause irrevocable data loss, verify host information above

*** Continue with migration? [y/N] y

2020-12-02T10:58:34.468-0800	INFO	migrate/migrate.go:173	applying migrations down
2020-12-02T10:58:34.475-0800	INFO	migrate/migrate.go:45	Start buffering 6/d experimentation_add_canceled_at_column
2020-12-02T10:58:34.486-0800	INFO	migrate/migrate.go:45	Read and execute 6/d experimentation_add_canceled_at_column
2020-12-02T10:58:34.507-0800	INFO	migrate/migrate.go:45	Finished 6/d experimentation_add_canceled_at_column (read 11.19196ms, ran 21.093481ms)
```

```
clutch=# select * from schema_migrations;
 version | dirty
---------+-------
       5 | f
(1 row)
```

Going back one more version to `4` should delete the `topology_cache` table.
```
➜  go run migrate.go -d -c ./testdata/clutch-config-test.yaml
2020-11-19T12:46:50.985-0800	INFO	migrate/migrate.go:93	using database	{"hostInfo": "clutch@0.0.0.0:5432"}
2020-11-19T12:46:50.985-0800	WARN	migrate/migrate.go:95	Migrating DOWN by ONE version from (5 -> 4) this migration has the potential to cause irrevocable data loss, verify host information above

*** Continue with migration? [y/N] y

2020-11-19T12:46:51.956-0800	INFO	migrate/migrate.go:129	Utilizing migration directory:	{"migrationDir": "/Users/mcutalo/gocode/src/github.com/lyft/clutch/backend/cmd/migrate/migrations"}
2020-11-19T12:46:51.956-0800	INFO	migrate/migrate.go:166	applying migrations down
2020-11-19T12:46:51.964-0800	INFO	migrate/migrate.go:45	Start buffering 5/d create_topology_cache_table
2020-11-19T12:46:51.977-0800	INFO	migrate/migrate.go:45	Read and execute 5/d create_topology_cache_table
2020-11-19T12:46:51.993-0800	INFO	migrate/migrate.go:45	Finished 5/d create_topology_cache_table (read 13.132322ms, ran 16.229795m
```

```
clutch=# select * from schema_migrations;
 version | dirty
---------+-------
       4 | f
(1 row
```

### CODE REVIEW

File: backend/cmd/migrate/migrations/000005_create_topology_cache_table.down.sql [hunk 1]
```diff
@@ -1 +1 @@
-DROP TABLE IF EXISTS topology_cache_data;
+DROP TABLE IF EXISTS topology_cache;
```
Comment (mcutalo88) [inline backend/cmd/migrate/migrations/000005_create_topology_cache_table.down.sql:1]: We renamed this table a while back, but this reference was not updated. Bundling it with this PR.

File: backend/cmd/migrate/migrate.go [hunk 2]
```diff
@@ -137,9 +139,32 @@ func (m *Migrator) Up() {
...
+
+func (m *Migrator) Down() {
+	msg := "Migrating DOWN by ONE version this migration has the potential to cause irrevocable data loss, verify host information above"
+	m.confirmWithUser(msg)
+
+	sqlMigrate := m.setupSqlMigrator()
+
+	// Migrate back by 1
+	m.log.Info("applying migrations down")
+	err := sqlMigrate.Steps(-1)
```
Comment (danielhochman) [inline backend/cmd/migrate/migrate.go:167]: let's have the confirmation include which step they're migrating from

Comment (mcutalo88) [reply backend/cmd/migrate/migrate.go]:
```
2020-12-02T10:58:28.455-0800	WARN	migrate/migrate.go:95	Migrating DOWN by ONE version from (6 -> 5) this migration has the potential to cause irrevocable data loss, verify host information above

*** Continue with migration? [y/N] y
```

**Review by danielhochman — APPROVED** [review]

File: backend/cmd/migrate/migrate.go [hunk 3]
```diff
@@ -33,6 +34,7 @@ func (m *MigrateFlags) Link() {
 	m.BaseFlags.Link()
 
 	flag.BoolVar(&m.Force, "f", false, "do not ask user for confirmation")
+	flag.BoolVar(&m.Down, "d", false, "migrates down by one version")
```
Comment (danielhochman) [inline backend/cmd/migrate/migrate.go:37]: let's make this a long-form arg -down if you don't mind
```suggestion
	flag.BoolVar(&m.Down, "down", false, "migrates down by one version")
```

File: backend/cmd/migrate/README.md [hunk 4]
```diff
@@ -10,3 +22,15 @@ go run migrate.go -template -c path/to/my/clutch-config.yaml
...
+## Migrating Down
+
+To migrate down supply the `-d` flag.
+
+A single invocation of the migrate down will only ever migrate one version down.
+If you need to migrate down multiple versions, invoke the tool until you reach the desired version.
+
+```
+cd backend/cmd/migrate
+go run migrate.go -template -d -c path/to/my/clutch-config.yaml
```
Comment (danielhochman) [inline backend/cmd/migrate/README.md:35]:
```suggestion
go run migrate.go -template -down -c path/to/my/clutch-config.yaml
```

File: backend/cmd/migrate/README.md [hunk 5]
```diff
@@ -10,3 +22,15 @@ go run migrate.go -template -c path/to/my/clutch-config.yaml
 Note: `migrate.go` accepts the same arguments as the main Clutch binary.
 
 Migrate also accepts an `-f` option to skip user confirmation of migration (useful for CI).
+
+## Migrating Down
+
+To migrate down supply the `-d` flag.
```
Comment (danielhochman) [inline backend/cmd/migrate/README.md:28]:
```suggestion
To migrate down supply the `-down` flag.
```

**Pull request merged**